gpflow/config/__config__.py

```diff
@@ -158,11 +158,11 @@ class Config:
     """
 
     int: type = field(default_factory=_default_int_factory)
-    float: type = field(default_factory=_default_float_factory)
```

Comment:
why this change? I would have thought it's nice to have int and float next to each other as they serve similar function

Reply:
The problem here is that once `float` has been defined in this list, MyPy interprets it as a type name when reading the subsequent signatures. As such `jitter` and `positive_minimum` have type `gpflow.config.__config__.Config.float`, which MyPy has already decided is not a type.

I can move the `int` to the bottom of the list to be placed next to the `float`.

Reply:
How about leaving them where they are, but using `__builtins__.float` as the type instead? then there is no implicit reliance on order (which someone else might inadvertently clean up again in the future!)

Reply:
i suspect the right approach is to rename `int` and `float` to e.g. `int_` or `int_type`, and leave names in the stdlib to the stdlib

Reply:
I'm happy to discuss renaming for a future release, for this PR I think it'd be good to go with `__builtins__.float` etc to make sure we don't implicitly depend on an order.

Reply:
I am having a bit of trouble with `__builtins__.float`. I will try a couple of other things and then update the PR.

Reply:
oh ok, my initial trials of `o: __builtins__.float = 1.0` worked fine in my python sandbox. is the difficulty coming from integrating with the rest of gpflow/your code base?

Reply:
I have not been able to get `__builtins__.float` to work with MyPy in this file. I have not tried running this over a larger collection of files. 

The main error I am seeing is this:
`    jitter: __builtins__.float = field(default_factory=_default_jitter_factory)
E   AttributeError: 'dict' object has no attribute 'float'
`

I have tried this with a few different versions of MyPy. I don't know why @joelberkeley-pio had different results with this approach. One approach that has worked (without requiring the ordering to change) is to introduce a new type alias above the class `Float = Union[float]`, and then use that type alias for entries in the `Config` class which are of `float` type. Would that be acceptable?

Reply:
i've just tried it in the gpflow file and works for me. `Float = Union[float]` is also fine by me

Reply:
+1 on `Float = Union[float]`

Reply:
(ideally with a short comment explaining why)